File: Code/C_Code/30.1.1_InfraredSensor/InfraredSensor.c
/**********************************************************************
* Filename    : InfraredSensor.c
* Description : Infrared Obstacle Avoidance Sensor control LED.
* Author      : www.freenove.com
* modification: 2022/4/20
**********************************************************************/
#include <wiringPi.h>
#include <stdio.h>
#define ledPin    0  	    //define the ledPin
#define sensorPin 1		//define the sensorPin

void  main(void)
{
    printf("Program is starting ... \n");
    wiringPiSetup();	//Initialize wiringPi.	
    pinMode(ledPin, OUTPUT); //Set ledPin to output
    pinMode(sensorPin, INPUT);//Set sensorPin to input
    while(1){
        if(digitalRead(sensorPin) == LOW){ // The sensor is blocked 
            digitalWrite(ledPin, HIGH);  //Make GPIO output HIGH level
            printf("led turned on >>>\n");    //Output information on terminal
        }
        else {							// The sensor is not blocked
            digitalWrite(ledPin, LOW);  //Make GPIO output LOW level
            printf("led turned off <<<\n");//Output information on terminal
        }
    }
}

File: Code/Python_Code/30.1.1_InfraredSensor/InfraredSensor.py
#!/usr/bin/env python3
#############################################################################
# Filename    : InfraredSensor.py
# Description : Infrared Obstacle Avoidance Sensor control LED.
# Author      : www.freenove.com
# modification: 2022/4/20
########################################################################
import RPi.GPIO as GPIO

ledPin = 11    # define ledPin
sensorPin = 12    # define bsensorPin

def setup():
    GPIO.setmode(GPIO.BOARD)      # use PHYSICAL GPIO Numbering
    GPIO.setup(ledPin, GPIO.OUT)   # set ledPin to OUTPUT mode
    GPIO.setup(sensorPin, GPIO.IN)    # set sensorPin to INPUT mode

def loop():
    while True:
        if GPIO.input(sensorPin)==GPIO.LOW: # The sensor is blocked
            GPIO.output(ledPin,GPIO.HIGH)   # turn on led
            print ('led turned on >>>')     # print information on terminal
        else : #  The sensor is not blocked
            GPIO.output(ledPin,GPIO.LOW) # turn off led	
            print ('led turned off <<<')	

def destroy(): 
    GPIO.cleanup()                     # Release GPIO resource

if __name__ == '__main__':     # Program entrance
    print ('Program is starting...')
    setup()
    try:
        loop()
    except KeyboardInterrupt:  # Press ctrl-c to end the program.
        destroy()
